### ISSUE TITLE
Kubernetes profile not propagated from bootstrap phase into ApplicationEnvironment

### ISSUE THREAD

**Issue by mecseid** [spring-cloud/spring-cloud-kubernetes#1261]
**Describe the bug**
With the latest version of Spring Cloud dependencies on Spring Boot 2.x tree, the `kubernetes` profile in the bootstrap phase is not propagated into the ApplicationServletEnvironment.
Because of that, when I want to check the `kubernetes` profile existence in the `logback-spring.xml` file, it will not find it and does not apply the custom settings.

I don't know if it is related only to Spring Cloud Kubernetes, or Spring Cloud Common too.

**Sample**
1. Create a maven project with Spring Boot 2.7.9, and Spring Cloud Release Train 2021.0.6.
2. Add the `spring-cloud-starter-kubernetes-fabric8-config` dependency.
3. Add a simple `kubernetes` Spring profile check into the `logback-spring.xml` file.
4. Start the application
5. The custom Logback setting will not take effect.

PS.: Looks like it works with Spring Boot 3 and the latest Spring Cloud Kubernetes version (with and without enabling the bootstrap participation).

**Comment by mecseid** [comment]
Yes, it works with the latest version, but we can't migrate from Spring Boot 2.x to 3.x one day to the next.
Although the workaround itself it's pretty simple (add `kubernetes` to `spring.profiles.active` property as CLI argument or as environment variable), I think it is not straightforward that you don't see `kubernetes` profile as active in the `ApplicationEnvironment`.

**Comment by wind57** [comment]
you should see it indeed, I can take a look at this one a bit closer, but I will need a minimal reproducible example. Not an explanation, but an example, like a github repo. thank you.

**Comment by mecseid** [comment]
Looks like I looked away, and the code works as expected (sorry for the non-existent issue @wind57).
I dug into the code and found that `spring.cloud.kubernetes.enabled` was set to `false `in `bootstrap.yaml`.
At first, I didn't find that because when I set a breakpoint into [AbstractKubernetesProfileEnvironmentPostProcessor.java#L62](https://github.com/spring-cloud/spring-cloud-kubernetes/blob/2.1.x/spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/profile/AbstractKubernetesProfileEnvironmentPostProcessor.java#L62), it said that kubernetes enabled in the bootstrap phase, but when Spring wanted to set the `ApplicationServletEnvironment`, it said `false`.

So I think the issue is closable.
